packages/nextjs/services/store/slices/ethPriceSlice.ts

```diff
@@ -12,8 +12,13 @@ export type TEthPriceSlice = {
 export const createEthPriceSlice: TAppSliceCreator<TEthPriceSlice> = set => ({
   ethPrice: defaultEthPriceState(),
   setEthPrice: (newValue: number): void =>
-    set((state): TAppStore => {
-      state.ethPriceSlice.ethPrice = newValue;
-      return state;
-    }),
+    set(
+      (state): TAppStore => ({
+        ...state,
```

Comment:
@carletex I think that this is not needed since the set method merges by default the state -> https://github.com/pmndrs/zustand/blob/main/docs/guides/immutable-state-and-merging.md

Reply:
Oh! Thanks for the long explanation!

If we refactor our slices to be at the same level then we have to worry about avoiding name collitions.

I think we can leave it now as you did it and then we can decide if there is a better solution.